src/app/admin/components/microschema-detail/mircoschema-detail.component.ts

```diff
@@ -87,9 +154,29 @@ export class MicroschemaDetailComponent implements OnInit, OnDestroy {
     delete() {
         this.microschema$
             .take(1)
-            .switchMap(microschema => this.adminSchemaEffects.deleteMicroschema(microschema.uuid))
+            .switchMap(microschema => this.schemaEffects.deleteMicroschema(microschema.uuid))
             .subscribe(() => this.router.navigate(['admin', 'microschemas']));
     }
+
+    onAssignmentChange(project: Project, isChecked: boolean) {
+        if (isChecked) {
+            this.microschema$
+                .take(1)
+                .map(microschema =>
+                    this.schemaEffects.assignEntityToProject('microschema', microschema.uuid, project.name)
+                )
+                .subscribe();
```

Comment:
Remove `map` and put the assignment in the `subscribe` method

Reply:
Done for schema and microschema.